### PR TITLE
Add time picker for Windows

### PR DESCRIPTION
**Pull request by rectified95**
# Summary
Implemented the 'time' mode for Windows. Now it's on par with iOS and Android.
Implemented the following properties: is24Hour, minuteInterval.
Added sample usage to the Example app. Above props can be toggled using a dropdown box.

Closes microsoft/react-native-windows#4695

[
windows_time_1
](url)
windows_time_2

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Windows |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: windows/DateTimePickerWindows/DateTimePickerView.idl [hunk 1]
```diff
@@ -8,4 +8,10 @@ namespace DateTimePicker {
         DateTimePickerView(Microsoft.ReactNative.IReactContext context);
         void UpdateProperties(Microsoft.ReactNative.IJSValueReader reader);
     };
+
+    [default_interface]
+    runtimeclass TimePickerView : Windows.UI.Xaml.Controls.TimePicker {
```
Comment (asklar) [inline windows/DateTimePickerWindows/DateTimePickerView.idl:13]: FYI once we'll start making modules work with WinUI 3 we'll need to change these things to inherit from `XAML_NAMESPACE.Controls.TimePicker` etc.

File: windows/DateTimePickerWindows/TimePickerView.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#include "TimePickerView.h"
+#include "TimePickerView.g.cpp"
+
+#include <winrt/Windows.Globalization.h>
+
+
+namespace winrt {
+    using namespace Microsoft::ReactNative;
+    using namespace Windows::Foundation;
+}
```
Comment (asklar) [inline windows/DateTimePickerWindows/TimePickerView.cpp:15]: is this actually being used?

File: windows/DateTimePickerWindows/TimePickerView.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            // User-selected time is represented as the number of 100ns slices elapsed from midnight (due to std::chrono). Conversion is required.
+            // As in iOS, the no. of miliseconds returned in the event will correspond to today's date, with the time value selected by the user.
+            // Additionally, JavaScript uses Jan 1, 1970 as its starting point for date calculations. std::chrono uses Jan 1, 1601. 
+
+            auto selectedTime = args.NewTime().GetTimeSpan().count();
+            selectedTime = selectedTime / chronoSliceInMillis;
+
+            auto nowInSeconds = winrt::clock::to_time_t(winrt::DateTime::clock::now());
+            auto daysSinceDawnOfTime = nowInSeconds / secondsPerDay;
+            auto tickCount = (daysSinceDawnOfTime * secondsPerDay) * 1000 + selectedTime;
```
Comment (asklar) [inline windows/DateTimePickerWindows/TimePickerView.cpp:95]: watch for overflows?

File: windows/DateTimePickerWindows/TimePickerView.cpp (hunk 2 above)
Comment (rectified95) [reply windows/DateTimePickerWindows/TimePickerView.cpp:15]: Yes, Microsoft::ReactNative is used in winrt::IReactContext;
Windows::Foundation is used in winrt::TimeSpan

Comment (rectified95) [reply windows/DateTimePickerWindows/TimePickerView.cpp:15]: Is there a reason we want to minimize the usage of winrt namespace aliases? I remember this being an issue in another PR.

Comment (asklar) [reply windows/DateTimePickerWindows/TimePickerView.cpp:15]: it obscures where types/apis come from. Also especially relevant when the same name is used for types in different namespaces (e.g. `Windows.UI.ApplicationModel.Activation.LaunchActivationEventArgs` versus `Microsoft.UI.ApplicationModel.Activation.LaunchActivationEventArgs`)

**Comment by rectified95** [comment]
@vonovak Can you review this one?

**Comment by vonovak** [comment]
@rectified95 please assign me as a reviewer when you're done here, thanks!

**Comment by rectified95** [comment]
@vonovak I don't have access right to do anything here :D That includes adding reviewers

**Comment by vonovak** [comment]
@rectified95 you have a pending invite to this repository since you did the last PR, can you please take a look and I'll resend it in case you don't have it. Thanks!

File: src/datetimepicker.windows.js [hunk 4]
```diff
@@ -50,13 +51,26 @@ export default class DateTimePickerWindows extends React.Component<WindowsNative
...
 
-    return (
-      <RNDateTimePickerWindows
-        {...props}
-        onChange={this._onChange}
-        timeZoneOffsetInSeconds={timezoneOffsetInSeconds}
-      />
-    );
+    // 'date' is the default mode
+    if (this.props.mode === 'date' || this.props.mode === undefined) {
```
Comment (vonovak) [inline src/datetimepicker.windows.js:55]: would you please rewrite this to use function component (with hooks) and also consider the case when `mode === null` (should be handled the same as `undefined` I think). Thanks!

**Comment by rectified95** [comment]
@vonovak Sorry for the late reply - I've been off this week. I'll rewrite the component using hooks on Monday.

About the invite: it got lost in my inbox, sorry about that. Now it says it's expired - can you, please, resend it? Thanks!

File: src/datetimepicker.windows.js (hunk 4 above)
Comment (rectified95) [reply src/datetimepicker.windows.js]: Rewrote as function component. 
Handled both undefined and null using the following comparison:
"mode **==** null"

**Comment by rectified95** [comment]
@vonovak 
Addressed your remarks.
Per my previous comment - kindly resend the collab invite for this repo - the previous one has already expired.
Thanks!

**Comment by vonovak** [comment]
@rectified95 I have re-sent the invite, can you please accept it? Thanks!

**Comment by rectified95** [comment]
@vonovak Accepted, thank you!

**Pull request merged**